game/__main__.py

```diff
@@ -84,21 +86,39 @@ def load_map(self):
             for key_layers in self.object_layers["key"]
         ]
 
+        self.interactable_list.extend(
+            Cabinet(loc) for loc in self.key_locations
+        )  # 🏄‍♂️some smooth sailing code
+
         self.enemy_list.extend(
             Enemy(self.wall_list, guard_location)
             for guard_location in self.guard_locations
         )
 
+    def handle_collision(self, key: int, modifiers: int):
+        original_pos = (self.player.center_x, self.player.center_y)
+        self.player.handle_user_input(key, modifiers)
+
+        if arcade.check_for_collision_with_list(self.player, self.wall_list):
+            self.player.center_x, self.player.center_y = original_pos
+
+        elif collisions := arcade.check_for_collision_with_list(
+            self.player, self.door_list
+        ):
+            if self.player.inventory.keys > 0:
+                self.player.inventory.keys -= 1
+                self.door_list.remove(collisions[0])
```

Comment:
we're removing doors? why not just have an open door texture. i'm guessing because it's easy to just get rid of the sprite? mmmm collision handling is much easier that way huh. ok, carry on

Reply:
I like your approach but time was a factor